### ISSUE TITLE
Plotting mesh edges only

### ISSUE THREAD

**Issue by os-gabe** [vispy/vispy#1650]
Hi,
This is a basic question but I'm trying to plot a mesh from an obj file but only show the edges and not the faces. I.e. I want a wireframe.

Is there an example of this that someone could point me to?

Thank you,

Gabe

**Comment by djhoese** [comment]
What does your code look like so far? Look through the examples in the root of this repository. I believe this should all be possible and that there is an example that loads from a `.obj` file. The `MeshVisual` that it should be using for drawing has options for specifying face and edge colors where I think you can ask for the face colors to be `None` or transparent.

**Comment by os-gabe** [comment]
Here is my code based on the examples:
```
import numpy as np
from vispy import scene
from vispy import app, io
import vispy


def mesh_viewer(fname, camera="arcball", bgcolor="black"):

    canvas = scene.SceneCanvas(keys="interactive", show=True, bgcolor=bgcolor)
    view = canvas.central_widget.add_view()

    view.camera = camera

    verts, faces, normals, nothin = io.read_mesh(fname)
    vc = np.zeros((len(verts), 4))
    fc = np.zeros((len(faces), 4))

    vc[:, 3] = 1
    vc[:, 0] = 1

    fc[:, 3] = 1
    fc[:, 1] = 1
    mesh = scene.visuals.Mesh(
        vertices=verts, faces=faces, face_colors=fc, vertex_colors=vc, mode="lines"
    )
    view.add(mesh)
    app.run()


fname = "<filename>"

print(vispy.config)
print(vispy.version_info)
print(vispy.sys_info())

mesh_viewer(fname)
```

The problem is that playing around with fc and vc above does not behave as expected.  I cannot a) make the faces invisible or b) make the edges visible

**Comment by os-gabe** [comment]
There is a mesh that can be used for testing here: https://people.sc.fsu.edu/~jburkardt/data/obj/alfa147.obj

**Comment by djhoese** [comment]
It's possible this hasn't been implemented yet. I really thought someone had worked on doing this recently, but can't find an issue or pull request talking about it. I really don't personally use mesh objects a lot in my own work so I'm not an expert. If you really need this you may have to dive in to the MeshVisual object and the MeshData object that it uses to see if you can figure it out. I'm sure it is possible, but right now the defaults assume you don't want to have transparent colors.

**Comment by os-gabe** [comment]
Thanks for looking into this @djhoese. I've since found that I can use the `mode` argument in `scene.visuals.Mesh` to achieve drawing of edges.

So if I want both faces and edges I can add a second visual and my `mesh_viewer` function would become something like:
```
def mesh_viewer(fname, camera="arcball", bgcolor="black"):

    canvas = scene.SceneCanvas(keys="interactive", show=True, bgcolor=bgcolor)
    view = canvas.central_widget.add_view()

    view.camera = camera

    verts, faces, normals, nothin = io.read_mesh(fname)

    vc1 = np.zeros((len(verts), 4))
    vc1[:, 3] = 0.5
    vc1[:, 1] = 1
    mesh1 = scene.visuals.Mesh(
        vertices=verts, faces=faces, vertex_colors=vc1, mode="triangles"
    )

    vc2 = np.zeros((len(verts), 4))
    vc2[:, 3] = 1
    vc2[:, 0] = 1
    mesh2 = scene.visuals.Mesh(vertices=verts, faces=faces, vertex_colors=vc2, mode="lines")
    view.add(mesh1)
    view.add(mesh2)
    app.run()
```

![Screenshot from 2019-06-10 11-40-06](https://user-images.githubusercontent.com/47365564/59218231-9a239080-8b74-11e9-9958-e78ba546e06d.png)

**Comment by GuillaumeFavelier** [comment]
Just for completeness as I don't know if it could be useful or not, I found that #1510 seemed to address this. Maybe you were thinking about this one @djhoese?

**Comment by djhoese** [comment]
Yep that's probably it. We'll see if a merge with master fixes that PR and where we can go from there.

**Comment by os-gabe** [comment]
Yes, #1510 addresses exactly what I was bringing up here. After my post above about using `mode="lines"`  I ran into the same problem talked about in #1510 where faces need to be passed in not as triangles but as an edge list. So my solution above does not work in general.

My fix was easy though. I just computed the edge array:
```
nf = len(faces)
mask = np.tile([0, 1, 1, 2, 2, 0], nf) + np.repeat(np.arange(0, nf * 3, 3), 6)
edges = faces.reshape(-1)[mask]
```
Then used a line visual with `connect="segments"`:
```
edge_visual = scene.visuals.Line(pos=verts[edges], connect="segments", color="white")
```

The edge computation above is for triangles but could easily be generalized to quads or polygons in general.

I'm not sure exactly what would have to happen to modify mesh visual to do this internally but it seems pretty straightforward.
